lib/experimental/new_embed.dart

```diff
@@ -232,10 +236,24 @@ class NewEmbed {
       ..setAnchorCorner(AnchorCorner.bottomLeft)
       ..setAnchorElement(menuButton._element.element);
     menu.listen('MDCMenu:selected', (e) {
-      if ((e as CustomEvent).detail['index'] == 0) {
-        _showTestCode = !_showTestCode;
-        showTestCodeCheckmark.toggleClass('hide', !_showTestCode);
-        tabController.setTabVisibility('test', _showTestCode);
+      final selectedIndex = (e as CustomEvent).detail['index'];
+      switch (selectedIndex) {
+        case 0: // Show Test Code
+          {
+            _showTestCode = !_showTestCode;
+            showTestCodeCheckmark.toggleClass('hide', !_showTestCode);
+            tabController.setTabVisibility('test', _showTestCode);
+            break;
+          }
+        case 1: // Editable test/solution
+          {
```

Comment:
same as above, this code block isn't necessary

Reply:
Removed unwanted blocks from switch/case